crml/staking/src/lib.rs

```diff
@@ -1573,47 +1482,66 @@ impl<T: Trait> Module<T> {
 		Self::select_validators().1
 	}
 
-	fn era_reward_payout() {
-		let validators = Self::current_elected();
-
-		// Pay the accumulated tx fee as rewards to all validators
-		let total_tx_fee_reward = CurrentEraFeeRewards::<T>::take();
-		Self::split_fee_rewards_evenly_to_all(&validators, total_tx_fee_reward);
-
-		let (total_payout, max_payout) = Self::current_total_payout(T::RewardCurrency::total_issuance());
-		let mut total_imbalance = T::RewardCurrency::burn(Zero::zero()); // hack to get new ImBalance with asset_id
-
-		let points = CurrentEraPointsEarned::take();
-		for (v, p) in validators.iter().zip(points.individual.into_iter()) {
-			if p != 0 {
-				let reward = Perbill::from_rational_approximation(p, points.total) * total_payout;
-				total_imbalance.subsume(Self::reward_validator(v, reward));
-			}
-		}
-
-		let total_payout = total_imbalance.peek();
-		let rest = max_payout.saturating_sub(total_payout);
-		Self::deposit_event(RawEvent::Reward(total_payout, rest));
+	/// Payout elected validators.
+	/// Builds a map of staker exposure and their preferred payment accounts.
+	/// A `StakerRewardPayment` impl will calculate the requisite payments.
+	fn era_reward_payout(elected_validator_stashes: Vec<T::AccountId>) {
+		let validator_commission_stake_map = elected_validator_stashes
+			.iter()
+			.map(|validator_stash| {
+				// Get a version of `Exposure` which maps to preferred payment account _instead of_ stash
+				let mut aggregate_stake = Self::stakers(validator_stash);
+				// get nominator payee accounts
+				aggregate_stake.others.iter_mut().for_each(|nominator_exposure| {
+					match Self::payee(&nominator_exposure.who) {
+						// `*nominator_exposure.who` is stash already
+						RewardDestination::Stash => {}
+						RewardDestination::Account(account) => nominator_exposure.who = account,
+						// TODO: this path requires two storage reads :/
```

Comment:
What should be the action?

Reply:
I think when we store a 'RewardDestination::Controller` we should store it as a `RewardDestination::Account(controller)`